### DISCUSSION TITLE
Should we rename to `rllm`?

### DISCUSSION THREAD

**Discussion by philpax** [rustformers/llama-rs#421] [Polls]
Something that's come up in conversation, as well as in other contexts, is that `llm` is not unique enough to identify this project. People refer to it as `rustformers/llm` or "Rustformers" - the former is overly verbose, and the latter is not strictly correct as we may develop other projects in the future. Additionally, Simon Willison, who has a fair bit of mindshare in the LLM-user community, has a project called [llm](https://github.com/simonw). I know I've been getting confused when I read his newsletter!

I'd like to gauge what people think of renaming this project specifically to `rllm` to give it a short, unique name that can be unambiguously used.

I considered `rrm`, but that rolls even worse off the tongue, and it risks sounding like a vaguely offensive riff on the R/L conflation in some Asian languages.

**Comment by danforbes** [comment]
Why I voted "No"
1. Inertia - This project has _months_ of history as `llm` as well as a number of dependent projects. IMO if the project is to be renamed it must be for _extremely_ good reason, which I don't think is the case (see below).
2. It's not an improvement - I don't think it's a better package name. Everyone knows what an LLM is, but people will ask themselves what a RLLM is. A regressive large language model? Furthermore, I think prevailing "wisdom" states that crate names do not contain the word "rust" and prefixing the crate name with an "r" is basically just a sneaky way of prefixing the name with the "idea" of Rust.
3. There are other options - An unambiguous way to refer to this project is "the Rust llm crate".
4. `llm` is clearly a good choice - If other people are using the same name, it should strengthen our resolve and not weaken it. Those other projects are in other ecosystems and do not effectively clash with this project.

**Reply by philpax** [reply]
Thanks for detailing your objection! No, seriously, I really appreciate it - it gives me much more context.

Here are my thoughts in response:
1. Months is not a long time in the grand scheme of things, especially since we haven't even cut an 0.2 yet. I figure if there's a non-zero potential of needing a rename, it's better to do it now than when we're further down the road and there are even more projects that might be broken by a rename.
2. I'm not sure about this - there's plenty of crates that are prefixed with `rust` or `r`. Using the data dump from [crates.io here](https://crates.io/data-access) and some quick Ruby: `ruby -e 'require "csv"; print(CSV.read("crates.csv", headers: true).by_col["name"].sort.select { |n| n.start_with? "rs-" }.len)'`, there are 130 crates starting with "rs-", 549 starting with "rust-", and 2831 starting with "ru". 
  
    I can't test for "r" because, well, that's a lot, but from a glance I can see a fair few like `rzip`, `rzw`, `rfetch`, `rlp`, and others. I don't think a 'r' prefix would be particularly out of the ordinary for someone familiar with the Rust ecosystem. `rllm` is also pretty unique right now - if you Google it, there are basically no relevant results, which means we'd likely shoot to the top of results.
3. Yes, that works, but it requires active effort on the speaker's part, instead of it "coming for free," as it were.
4. Sure, but that will only get more confusing over time, not less. As more and more things refer to LLMs, the area of awareness we occupy will shrink.

As a useful piece of information: a lot of my concerns are based around the CLI, which is called `llm`. Simon Willison's LLM tool is *also* called `llm`, and it's not unlikely at all that someone might have both installed, especially if they're keeping up with the field. It'd be nice to avoid that conflict.

It'd also be really nice to have a term that people can search for and get a unique result for, as is the case when you search for "Actix" or "Tokio" (but I will admit that the latter is very easily confused with Tokyo).

How do you feel about "rullm"? That should address the concern you have about the "r" potentially referring to something else, and should make it more obvious that it's specifically a Rust thing.

**Comment by philpax** [comment]
There's no clear consensus here, so I'm going to close this and keep it `llm`, and suggest `llm.rs` where disambiguation is required. (I just bought `llm.rs` and am trying to set it up as a redirect to this repository, but I don't know Serbian, so... 😅)

**Reply by ggerganov** [reply]
`llm.rs` sounds great 😄
